hub/api/datasetview.py

```diff
@@ -18,53 +13,165 @@ def __init__(
+            return DatasetView(
+                dataset=self.dataset, num_samples=num, offset=self.offset + ofs
+            )
 
         elif isinstance(slice_, str):
-            subpath = slice_ if slice_.startswith("/") else "/" + slice_     # return TensorView object
-            # add slice for original Dataset range
-            return TensorView(dataset=self.dataset, subpath=subpath, slice_=slice(self.offset, self.offset + self.num_samples))
+            subpath = slice_ if slice_.startswith("/") else "/" + slice_
+            # return TensorView object
+            if self.num_samples == 1:
+                slice_ = self.offset
+            else:
+                slice_ = slice(self.offset, self.offset + self.num_samples)
+
+            if subpath in self._tensors.keys():
+                return TensorView(dataset=self.dataset, subpath=subpath, slice_=slice_)
+            else:
+                d = {}
+                post_subpath = subpath if subpath.endswith("/") else subpath + "/"
+                for key in self._tensors.keys():
+                    if key.startswith(post_subpath):
+                        suffix_key = key[len(post_subpath) :]
+                    else:
+                        continue
+                    split_key = suffix_key.split("/")
+                    cur = d
+                    for i in range(len(split_key) - 1):
+                        if split_key[i] in cur.keys():
+                            cur = cur[split_key[i]]
+                        else:
+                            cur[split_key[i]] = {}
+                            cur = cur[split_key[i]]
+                    cur[split_key[-1]] = TensorView(
+                        dataset=self, subpath=key, slice_=slice_
+                    )
+
+                if len(d) == 0:
+                    raise KeyError(f"Key {subpath} was not found in dataset")
+                return d
 
-        elif isinstance(slice_, tuple):        # return tensor view object
+        elif isinstance(slice_, tuple):
+            # return tensor view object
             subpath, slice_ = slice_split_tuple(slice_)
 
             if len(slice_) == 0:
-                slice_ = None
-            elif isinstance(slice_[0], int):
-                if slice_[0] >= self.num_samples:
-                    raise IndexError('index out of bounds for dimension with length {}'.format(self.num_samples))
-                ls = list(slice_)
-                ls[0] += self.offset
-                slice_ = tuple(ls)
-            elif isinstance(slice_[0], slice):
-                num, ofs = slice_extract_info(slice_[0], self.num_samples)
-                ls = list(slice_)
-                ls[0] = slice(self.offset + ofs, self.offset + ofs + num)
-                slice_ = tuple(ls)
-            return TensorView(dataset=self.dataset, subpath=subpath, slice_=slice_)
+                if self.num_samples == 1:
```

Comment:
Too Long function, without documentation

Reply:
@AbhinavTuli I think you should respond to this.

Reply:
Fixing and adding documentation now.